### PR TITLE
#150 Bugfix declaration of queue needs to be done before start of con…

### PR DESCRIPTION
**Pull request by Gr33nbl00d**
…sume

---
This change is [Reviewable](https://reviewable.io/reviews/reinhapa/rabbitmq-cdi/169)

### CODE REVIEW

**Comment by reinhapa** [comment]
Is there a way to extend an existing test for this regression?

**Comment by Gr33nbl00d** [comment]
I have added a test for this

File: src/test/java/net/reini/rabbitmq/cdi/ConsumerHolderTest.java [hunk 1]
```diff
@@ -89,6 +91,21 @@ void activateAndDeactivate() throws IOException, TimeoutException {
     verify(channelMock).close();
   }
 
+  @Test
+  void testActivationOrder() throws IOException {
+    sut = new ConsumerHolder(eventConsumerMock, "queue", false, PREFETCH_COUNT,
+        consumerChannelFactoryMock, declarationsListMock, declarerRepositoryMock);
+    Assertions.assertEquals("queue", sut.getQueueName());
```
Comment (reinhapa) [inline src/test/java/net/reini/rabbitmq/cdi/ConsumerHolderTest.java:98]: Change those to static imports

File: src/test/java/net/reini/rabbitmq/cdi/ConsumerHolderTest.java [hunk 2]
```diff
@@ -89,6 +91,21 @@ void activateAndDeactivate() throws IOException, TimeoutException {
...
+    when(consumerChannelFactoryMock.createChannel()).thenReturn(channelMock);
+    sut.activate();
```
Comment (reinhapa) [inline src/test/java/net/reini/rabbitmq/cdi/ConsumerHolderTest.java:100]: In order to have at least one  "assert" you could write:
`assertDoesNotThrow(sut::activate)`

File: src/test/java/net/reini/rabbitmq/cdi/ConsumerHolderTest.java (hunk 1 above)
Comment (Gr33nbl00d) [reply src/test/java/net/reini/rabbitmq/cdi/ConsumerHolderTest.java:98]: fixed

File: src/test/java/net/reini/rabbitmq/cdi/ConsumerHolderTest.java (hunk 2 above)
Comment (Gr33nbl00d) [reply src/test/java/net/reini/rabbitmq/cdi/ConsumerHolderTest.java:100]: fixed

**Pull request merged**